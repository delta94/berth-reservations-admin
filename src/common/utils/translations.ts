import {
  AdditionalProductTaxEnum,
  CustomerGroup,
  OrderStatus,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../@types/__generated__/globalTypes';
import { formatPercentage } from './format';

export const getPeriodTKey = (period: PeriodType) => {
  switch (period) {
    case PeriodType.MONTH:
      return 'common.periodTypes.MONTH';
    case PeriodType.SEASON:
      return 'common.periodTypes.SEASON';
    case PeriodType.YEAR:
      return 'common.periodTypes.YEAR';

    default:
      return period;
  }
};

export const getProductServiceTKey = (productService: ProductServiceType) => {
  switch (productService) {
    case ProductServiceType.DINGHY_PLACE:
      return 'common.terminology.dinghyPlace';
    case ProductServiceType.ELECTRICITY:
      return 'common.terminology.electricity';
    case ProductServiceType.GATE:
      return 'common.terminology.gate';
    case ProductServiceType.LIGHTING:
      return 'common.terminology.lighting';
    case ProductServiceType.MOORING:
      return 'common.terminology.mooring';
    case ProductServiceType.PARKING_PERMIT:
      return 'common.terminology.parkingPermit';
    case ProductServiceType.SUMMER_STORAGE_FOR_DOCKING_EQUIPMENT:
      return 'common.terminology.dockingEquipmentSummerStorage';
    case ProductServiceType.SUMMER_STORAGE_FOR_TRAILERS:
      return 'common.terminology.trawlerSummerStorage';
    case ProductServiceType.WASTE_COLLECTION:
      return 'common.terminology.wasteCollection';
    case ProductServiceType.WATER:
      return 'common.terminology.water';

    default:
      return productService;
  }
};

export const getOrderStatusTKey = (orderStatus: OrderStatus): string => {
  switch (orderStatus) {
    case OrderStatus.CANCELLED:
      return 'common.orderStatus.cancelled';
    case OrderStatus.EXPIRED:
      return 'common.orderStatus.expired';
    case OrderStatus.PAID:
      return 'common.orderStatus.paid';
    case OrderStatus.REJECTED:
      return 'common.orderStatus.rejected';
    case OrderStatus.WAITING:
      return 'common.orderStatus.waiting';

    default:
      return orderStatus;
  }
};

export const getProductTax = (tax: AdditionalProductTaxEnum, locale: string) => {
  switch (tax) {
    case AdditionalProductTaxEnum.TAX_10_00:
      return formatPercentage(10, locale);
    case AdditionalProductTaxEnum.TAX_24_00:
      return formatPercentage(24, locale);

    default:
      return tax;
  }
};

export const getPriceUnits = (unit: PriceUnits) => {
  switch (unit) {
    case PriceUnits.AMOUNT:
      return 'â‚¬';
    case PriceUnits.PERCENTAGE:
      return '%';

    default:
      return unit;
  }
};

export const getCustomerGroupKey = (customerGroup: CustomerGroup | null): string => {
  return customerGroup ? `common.customerGroups.${customerGroup}` : 'common.customerGroups.PRIVATE';
};
