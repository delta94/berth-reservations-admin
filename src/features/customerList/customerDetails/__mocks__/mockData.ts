import { CustomerListApplication, CustomerListBerthLeases, CustomerListBill, CustomerListBoat } from '../../types';
import { CustomerGroup } from '../../../../@types/__generated__/globalTypes';
import { CustomerDetailsProps } from '../CustomerDetails';

export const customerListEntry: Omit<
  CustomerDetailsProps,
  'berths' | 'winterStoragePlaces' | 'boats' | 'applications' | 'bills'
> = {
  name: 'Mikko Matias Mallikas',
  address: 'Telakkakatu 1 A 10',
  postalCode: '00100',
  city: 'Helsinki',
  phone: '+358 040 123 4567',
  email: 'mikko.mallikas@meri.fi',
  customerGroup: CustomerGroup.PRIVATE,
  comment:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean malesuada id est ut pellentesque. ' +
    'Vivamus quis maximus sem. Ut auctor vestibulum mattis. Vestibulum mollis diam convallis ligula consequat ' +
    'sagittis.',
};

export const customerListWinterStoragePlaces = [{ id: '123', title: 'Rajasaari' }];

export const customerListBoats: CustomerListBoat[] = [{ id: '123', name: 'Cama la Yano' }];

export const customerListApplications: CustomerListApplication[] = [{ id: '123', createdAt: '2019-12-01' }];

export const customerListBills: CustomerListBill[] = [{ id: '123', date: '2020-01-21' }];

export const customerListBerthLeases: CustomerListBerthLeases[] = [
  { id: '123', title: 'Pursilahdenranta B31' },
  { id: '321', title: 'Str√∂msinlahdenranta B31' },
];
