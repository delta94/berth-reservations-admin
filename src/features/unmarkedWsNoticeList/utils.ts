import { ApplicationStatus } from '../../@types/__generated__/globalTypes';
import {
  UNMARKED_WINTER_STORAGE_APPLICATIONS,
  UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node_winterStorageAreaChoices as CHOICE,
} from './__generated__/UNMARKED_WINTER_STORAGE_APPLICATIONS';

interface UnmarkedWinterStorageChoice {
  winterStorageAreaName: string;
  winterStorageArea: string;
}

export type UnmarkedWinterStorageNotice = {
  id: string;
  queue: number | null;
  status: ApplicationStatus;
  createdAt: string;
  municipality: string;
  customerId?: string;
  boatType?: string | null;
  boatRegistrationNumber: string;
  boatWidth: number;
  boatLength: number;
  boatName: string;
  boatModel: string;
  choice: UnmarkedWinterStorageChoice;
};

export const getUnmarkedWinterStorageNotice = (
  data: UNMARKED_WINTER_STORAGE_APPLICATIONS | undefined
): UnmarkedWinterStorageNotice[] => {
  if (!data || !data.winterStorageApplications) {
    return [];
  }

  return data.winterStorageApplications.edges.reduce<UnmarkedWinterStorageNotice[]>((acc, edge) => {
    if (!edge || !edge.node) {
      return acc;
    }
    const application = edge.node;

    const choices = application.winterStorageAreaChoices;
    const choice: UnmarkedWinterStorageChoice =
      choices !== null && choices.length > 0 && choices[0] !== null
        ? {
            winterStorageArea: (choices[0] as CHOICE).winterStorageArea || '',
            winterStorageAreaName: (choices[0] as CHOICE).winterStorageArea || '',
          }
        : {
            winterStorageArea: 'Virhe',
            winterStorageAreaName: 'Virhe',
          };

    const applicationData: UnmarkedWinterStorageNotice = {
      id: application.id,
      queue: null, // TODO: populate from query once available
      status: application.status,
      createdAt: application.createdAt,
      municipality: application.municipality,
      customerId: application.customer?.id,
      boatType: data.boatTypes?.find(({ id }) => id === application.boatType)?.name,
      boatRegistrationNumber: application.boatRegistrationNumber,
      boatWidth: application.boatWidth,
      boatLength: application.boatLength,
      boatName: application.boatName,
      boatModel: application.boatModel,
      choice,
    };
    return [...acc, applicationData];
  }, []);
};
