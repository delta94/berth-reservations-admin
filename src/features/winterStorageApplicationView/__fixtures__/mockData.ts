import {
  INDIVIDUAL_WINTER_STORAGE_APPLICATION,
  INDIVIDUAL_WINTER_STORAGE_APPLICATION_boatTypes as BOAT_TYPES,
  INDIVIDUAL_WINTER_STORAGE_APPLICATION_winterStorageApplication as WINTER_STORAGE_APPLICATION,
} from '../__generated__/INDIVIDUAL_WINTER_STORAGE_APPLICATION';
import { ApplicationStatus, BerthApplicationLanguage } from '../../../@types/__generated__/globalTypes';

export const mockData: INDIVIDUAL_WINTER_STORAGE_APPLICATION & {
  winterStorageApplication: WINTER_STORAGE_APPLICATION;
  boatTypes: BOAT_TYPES[];
} = {
  winterStorageApplication: {
    id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTox',
    firstName: 'test',
    lastName: 'test',
    address: 'test',
    municipality: 'helsinki',
    zipCode: '00100',
    phoneNumber: '555555',
    email: 'test@test.com',
    businessId: '',
    companyName: '',
    language: BerthApplicationLanguage.EN,
    customer: null,
    createdAt: '2019-05-06T10:54:21.746387+00:00',
    boatType: '3',
    boatRegistrationNumber: 'test',
    boatWidth: 1,
    boatLength: 1,
    boatName: 'test',
    boatModel: 'test',
    acceptBoatingNewsletter: true,
    acceptFitnessNews: true,
    acceptLibraryNews: true,
    acceptOtherCultureNews: true,
    status: ApplicationStatus.PENDING,
    winterStorageAreaChoices: [
      {
        winterStorageArea: null,
        priority: 1,
        winterStorageAreaName: 'Iso-Sarvasto',
        __typename: 'WinterStorageAreaChoiceType',
      },
      {
        winterStorageArea: null,
        priority: 2,
        winterStorageAreaName: 'Rajasaari',
        __typename: 'WinterStorageAreaChoiceType',
      },
    ],
    applicationCode: '1234',
    __typename: 'WinterStorageApplicationNode',
  },
  boatTypes: [
    {
      id: '1',
      name: 'Jollavene',
      __typename: 'BoatTypeType',
    },
    {
      id: '2',
      name: 'Soutuvene',
      __typename: 'BoatTypeType',
    },
    {
      id: '3',
      name: 'Perämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '4',
      name: 'Sisäperämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '5',
      name: 'Keskimoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '6',
      name: 'Purjevene / moottoripursi',
      __typename: 'BoatTypeType',
    },
    {
      id: '7',
      name: 'Troolari',
      __typename: 'BoatTypeType',
    },
    {
      id: '8',
      name: 'Suuri alus (yli 20t)',
      __typename: 'BoatTypeType',
    },
  ],
};
