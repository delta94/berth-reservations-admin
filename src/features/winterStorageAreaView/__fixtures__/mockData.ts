import { INDIVIDUAL_WINTER_STORAGE_AREA } from '../__generated__/INDIVIDUAL_WINTER_STORAGE_AREA';
import { LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: INDIVIDUAL_WINTER_STORAGE_AREA = {
  winterStorageArea: {
    properties: {
      name: 'WS Area 1',
      zipCode: '00100',
      municipality: 'Helsinki',
      streetAddress: '',
      wwwUrl: '',
      imageFile: null,
      maps: [],
      sections: {
        edges: [
          {
            node: {
              id: 'V2ludGVyU3RvcmFnZVNlY3Rpb25Ob2RlOjI1OTgyMTEyLTllNDktNGUwNy1hYmVjLTZhMTQyMzcyZmZjOQ==',
              properties: {
                identifier: '-',
                electricity: true,
                water: true,
                gate: false,
                summerStorageForBoats: false,
                summerStorageForTrailers: false,
                summerStorageForDockingEquipment: false,
                leases: {
                  __typename: 'WinterStorageLeaseNodeConnection',
                  edges: [],
                },
                places: {
                  edges: [
                    {
                      node: {
                        id: 'V2ludGVyU3RvcmFnZVBsYWNlTm9kZTo0YzUzZTY3NC0zMmFjLTRkNTEtYmQzMC0wNTcyMWNjMjUyMTE=',
                        number: 1,
                        width: 3,
                        length: 5,
                        isActive: true,
                        leases: {
                          edges: [
                            {
                              node: {
                                id: 'V2ludGVyU3RvcmFnZUxlYXNlTm9kZTo3ZTU0NjdmMy1jZGJhLTQzMmEtODdiZi05ODBiZTI1ZGFjNzg=',
                                startDate: '2020-09-15',
                                endDate: '2021-06-10',
                                status: LeaseStatus.DRAFTED,
                                isActive: false,
                                application: {
                                  createdAt: '2020-07-17T10:52:57.079036+00:00',
                                  customer: {
                                    id: 'UHJvZmlsZU5vZGU6NWNjYzgyNDUtNmRiOS00YTRiLWI5NTEtNWYxNDQ5YTY5NzY2',
                                    firstName: 'Heikki',
                                    lastName: 'Kinnunen',
                                    __typename: 'ProfileNode',
                                  },
                                  id: 'foo',
                                  __typename: 'WinterStorageApplicationNode',
                                },
                                __typename: 'WinterStorageLeaseNode',
                              },
                              __typename: 'WinterStorageLeaseNodeEdge',
                            },
                          ],
                          __typename: 'WinterStorageLeaseNodeConnection',
                        },
                        __typename: 'WinterStoragePlaceNode',
                      },
                      __typename: 'WinterStoragePlaceNodeEdge',
                    },
                    {
                      node: {
                        id: 'V2ludGVyU3RvcmFnZVBsYWNlTm9kZTo3OTkzYzAxOC0zMTMwLTRjNTItYjA3Ni1mMGZlN2Q5YjRmYTE=',
                        number: 2,
                        width: 3,
                        length: 5,
                        isActive: true,
                        leases: {
                          edges: [
                            {
                              node: {
                                id: 'V2ludGVyU3RvcmFnZUxlYXNlTm9kZTpjOWNjODAyNi1hZTkzLTRjMTUtYmJiNy1kM2Q1ZjE4ZjViYjA=',
                                startDate: '2020-09-15',
                                endDate: '2021-06-10',
                                status: LeaseStatus.DRAFTED,
                                isActive: false,
                                application: null,
                                __typename: 'WinterStorageLeaseNode',
                              },
                              __typename: 'WinterStorageLeaseNodeEdge',
                            },
                          ],
                          __typename: 'WinterStorageLeaseNodeConnection',
                        },
                        __typename: 'WinterStoragePlaceNode',
                      },
                      __typename: 'WinterStoragePlaceNodeEdge',
                    },
                  ],
                  __typename: 'WinterStoragePlaceNodeConnection',
                },
                __typename: 'WinterStorageSectionProperties',
              },
              __typename: 'WinterStorageSectionNode',
            },
            __typename: 'WinterStorageSectionNodeEdge',
          },
        ],
        __typename: 'WinterStorageSectionNodeConnection',
      },
      __typename: 'WinterStorageAreaProperties',
    },
    __typename: 'WinterStorageAreaNode',
  },
};
