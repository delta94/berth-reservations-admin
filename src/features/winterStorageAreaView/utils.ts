import {
  INDIVIDUAL_WINTER_STORAGE_AREA,
  INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections as SECTIONS,
  INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node as SECTION,
  INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties as SECTION_PROPERTIES,
  INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_leases as LEASES,
  // eslint-disable-next-line max-len
  INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges as WINTER_STORAGE_PLACES,
  INDIVIDUAL_WINTER_STORAGE_AREA_winterStorageArea_properties_sections_edges_node_properties_places_edges_node as PLACE,
} from './__generated__/INDIVIDUAL_WINTER_STORAGE_AREA';
import {
  WinterStorageArea,
  Map,
  WinterStoragePlace,
  WinterStorageSection,
  Lease,
  MarkedWinterStorage,
  UnmarkedWinterStorage,
} from './types';

const getNumberOfCustomersForPlace = (place: PLACE): number => {
  const isActive = place?.leases?.edges?.find((edge) => edge?.node?.isActive);
  return isActive ? 1 : 0;
};

const getNumberOfCustomersForSection = (section: SECTION) => {
  const countPerPlace =
    section?.properties?.places.edges.map((edge) => (edge?.node ? getNumberOfCustomersForPlace(edge.node) : 0)) || [];

  return countPerPlace.reduce((a, b) => {
    return a + b;
  }, 0);
};

const getNumberOfCustomers = (sections: SECTIONS) => {
  const countPerSection =
    sections.edges.map((edge) => (edge?.node ? getNumberOfCustomersForSection(edge.node) : 0)) || [];

  return countPerSection.reduce((a, b) => {
    return a + b;
  }, 0);
};

export const getIndividualWinterStorageArea = (
  data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined
): WinterStorageArea | null => {
  if (!data || !data.winterStorageArea || !data.winterStorageArea.properties) {
    return null;
  }
  const { name, zipCode, municipality, streetAddress, wwwUrl, imageFile, sections } = data.winterStorageArea.properties;

  const maps = data.winterStorageArea.properties.maps.reduce<Map[]>((acc, map) => {
    if (map !== null) {
      return acc.concat({
        id: map.id,
        url: map.url,
      });
    }
    return acc;
  }, []);

  const hasService = (service: keyof SECTION_PROPERTIES): boolean => {
    return (
      data.winterStorageArea?.properties?.sections.edges.some((edge) => edge?.node?.properties?.[service]) ?? false
    );
  };

  return {
    name: name ?? '',
    zipCode,
    municipality,
    streetAddress,
    wwwUrl,
    imageFile,
    maps,
    electricity: hasService('electricity'),
    water: hasService('water'),
    gate: hasService('gate'),
    summerStorageForBoats: hasService('summerStorageForBoats'),
    summerStorageForTrailers: hasService('summerStorageForTrailers'),
    summerStorageForDockingEquipment: hasService('summerStorageForDockingEquipment'),
    numberOfCustomers: getNumberOfCustomers(sections),
  };
};

const getLeases = (leases: LEASES): Lease[] => {
  return leases?.edges.reduce<Lease[]>((acc, leaseEdge) => {
    if (!leaseEdge?.node?.application?.customer) {
      return acc;
    }
    const { id, status, startDate, endDate, isActive } = leaseEdge.node;
    return [
      ...acc,
      {
        id,
        customer: {
          id: leaseEdge.node.application.customer.id,
          firstName: leaseEdge.node.application.customer.firstName,
          lastName: leaseEdge.node.application.customer.lastName,
        },
        status,
        applicationId: leaseEdge.node.application.id,
        applicationDate: leaseEdge.node.application.createdAt,
        startDate,
        endDate,
        isActive,
      },
    ];
  }, []);
};

const getLeasesForPlaces = (places: WINTER_STORAGE_PLACES): Lease[] => {
  return places?.node?.leases ? getLeases(places.node.leases) : [];
};

const getWinterStorageLeasesFromSectionProperties = (sectionProperties: SECTION_PROPERTIES) => {
  return sectionProperties?.leases ? getLeases(sectionProperties.leases) : [];
};

const getWinterStoragePlacesFromSectionProperties = (sectionProperties: SECTION_PROPERTIES) => {
  return (
    sectionProperties.places.edges.reduce<WinterStoragePlace[]>((acc, placeEdge) => {
      if (!placeEdge?.node) {
        return acc;
      }
      const leases = getLeasesForPlaces(placeEdge);
      return [
        ...acc,
        {
          id: placeEdge.node.id,
          identifier: sectionProperties.identifier,
          number: placeEdge.node.number,
          width: placeEdge.node.width,
          length: placeEdge.node.length,
          isActive: placeEdge.node.isActive,
          leases,
        },
      ];
    }, []) ?? []
  );
};

export const getWinterStoragePlaces = (data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined): WinterStoragePlace[] => {
  if (!data?.winterStorageArea?.properties) {
    return [];
  }
  return data?.winterStorageArea?.properties?.sections.edges
    .reduce<SECTION_PROPERTIES[]>((acc, sectionEdge) => {
      if (!sectionEdge?.node?.properties) {
        return acc;
      }
      return [...acc, sectionEdge.node.properties];
    }, [])
    .map<WinterStoragePlace[]>(getWinterStoragePlacesFromSectionProperties)
    .reduce<WinterStoragePlace[]>((acc, arr) => [...acc, ...arr], []);
};

export const getUnmarkedWinterStorageLeases = (data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined): Lease[] => {
  if (!data?.winterStorageArea?.properties) {
    return [];
  }
  return data?.winterStorageArea?.properties?.sections.edges
    .reduce<SECTION_PROPERTIES[]>((acc, sectionEdge) => {
      if (!sectionEdge?.node?.properties) {
        return acc;
      }
      return [...acc, sectionEdge.node.properties];
    }, [])
    .map<Lease[]>(getWinterStorageLeasesFromSectionProperties)
    .reduce<Lease[]>((acc, arr) => [...acc, ...arr], []);
};

const isUnmarkedSection = (section: SECTION) => {
  if (!section?.properties?.leases?.edges) return false;
  return section.properties?.leases?.edges?.length > 0;
};

export const getWinterStorageSections = (
  data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined,
  unmarked: boolean
): WinterStorageSection[] => {
  if (!data?.winterStorageArea?.properties) {
    return [];
  }
  return data.winterStorageArea.properties.sections.edges.reduce<WinterStorageSection[]>((acc, sectionEdge) => {
    if (!sectionEdge?.node?.properties?.identifier) {
      return acc;
    }
    if (unmarked ? !isUnmarkedSection(sectionEdge.node) : isUnmarkedSection(sectionEdge.node)) {
      return acc;
    }
    return [...acc, { identifier: sectionEdge.node.properties.identifier }];
  }, []);
};

export const getMarkedWinterStorage = (
  data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined
): MarkedWinterStorage | undefined => {
  const places = getWinterStoragePlaces(data);
  const sections = getWinterStorageSections(data, false);

  if (places.length === 0) return undefined;
  return {
    places,
    sections,
  };
};

export const getUnmarkedWinterStorage = (
  data: INDIVIDUAL_WINTER_STORAGE_AREA | undefined
): UnmarkedWinterStorage | undefined => {
  const leases = getUnmarkedWinterStorageLeases(data);

  if (leases.length === 0) return undefined;
  return {
    leases,
  };
};
