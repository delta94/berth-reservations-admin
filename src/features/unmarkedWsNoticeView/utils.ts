import {
  UNMARKED_WINTER_STORAGE_NOTICE_winterStorageApplication as WINTER_STORAGE_APPLICATION,
  UNMARKED_WINTER_STORAGE_NOTICE_boatTypes as BOAT_TYPES,
} from './__generated__/UNMARKED_WINTER_STORAGE_NOTICE';
import { getApplicantDetails } from '../applicationView/utils';
import { UNMARKED_WINTER_STORAGE_APPLICATIONS_winterStorageApplications_edges_node_winterStorageAreaChoices as CHOICE } from '../unmarkedWsNoticeList/__generated__/UNMARKED_WINTER_STORAGE_APPLICATIONS';
import { UnmarkedWsNoticeDetailsProps } from '../../common/unmarkedWsNoticeDetails/UnmarkedWsNoticeDetails';

interface UnmarkedWinterStorageChoice {
  winterStorageAreaName: string;
  winterStorageArea: string;
}

export const getWinterStorageApplicationDetailsData = (
  winterStorageApplication: WINTER_STORAGE_APPLICATION,
  boatTypes: BOAT_TYPES[]
): UnmarkedWsNoticeDetailsProps & Required<Pick<UnmarkedWsNoticeDetailsProps, 'applicant'>> => {
  const choices = winterStorageApplication.winterStorageAreaChoices;
  const choice: UnmarkedWinterStorageChoice =
    choices !== null && choices.length > 0 && choices[0] !== null
      ? {
          winterStorageArea: (choices[0] as CHOICE).winterStorageArea || '',
          winterStorageAreaName: (choices[0] as CHOICE).winterStorageArea || '',
        }
      : {
          winterStorageArea: 'Virhe',
          winterStorageAreaName: 'Virhe',
        };

  return {
    ...winterStorageApplication,
    customerId: winterStorageApplication.customer?.id,
    applicant: getApplicantDetails(winterStorageApplication),
    queue: null,
    choice,
    boatType: boatTypes.find(({ id }) => id === winterStorageApplication.boatType)?.name,
    summaryInformation: {
      applicationCode: winterStorageApplication.applicationCode,
      acceptBoatingNewsletter: winterStorageApplication.acceptBoatingNewsletter,
      acceptFitnessNews: winterStorageApplication.acceptFitnessNews,
      acceptLibraryNews: winterStorageApplication.acceptLibraryNews,
      acceptOtherCultureNews: winterStorageApplication.acceptOtherCultureNews,
    },
  };
};
