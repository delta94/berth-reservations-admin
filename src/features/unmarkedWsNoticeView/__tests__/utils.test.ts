import { mockData } from '../__fixtures__/mockData';
import { getWinterStorageApplicationDetailsData } from '../utils';

describe('unmarkedWsNoticeView utils', () => {
  describe('getWinterStorageApplicationDetailsData', () => {
    it('parses data correctly', () => {
      const winterStorageApplicationDetailsData = getWinterStorageApplicationDetailsData(
        mockData.winterStorageApplication,
        mockData.boatTypes
      );
      expect(winterStorageApplicationDetailsData).toMatchSnapshot();
    });
  });
});
