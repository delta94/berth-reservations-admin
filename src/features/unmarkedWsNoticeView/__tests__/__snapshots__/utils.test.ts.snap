// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unmarkedWsNoticeView utils getWinterStorageApplicationDetailsData parses data correctly 1`] = `
Object {
  "__typename": "WinterStorageApplicationNode",
  "acceptBoatingNewsletter": true,
  "acceptFitnessNews": true,
  "acceptLibraryNews": true,
  "acceptOtherCultureNews": true,
  "address": "test",
  "applicant": Object {
    "customerGroup": "PRIVATE",
    "firstName": "test",
    "language": "ENGLISH",
    "lastName": "test",
    "primaryAddress": Object {
      "address": "test",
      "city": "helsinki",
      "postalCode": "00100",
    },
    "primaryEmail": "test@test.com",
    "primaryPhone": "555555",
  },
  "applicationCode": "1234",
  "boatLength": 1,
  "boatModel": "test",
  "boatName": "test",
  "boatRegistrationNumber": "test",
  "boatType": "Per√§moottorivene",
  "boatWidth": 1,
  "businessId": "",
  "choices": Array [
    Object {
      "priority": 1,
      "winterStorageArea": "",
      "winterStorageAreaName": "Iso-Sarvasto",
    },
    Object {
      "priority": 2,
      "winterStorageArea": "",
      "winterStorageAreaName": "Rajasaari",
    },
  ],
  "companyName": "",
  "createdAt": "2019-05-06T10:54:21.746387+00:00",
  "customer": null,
  "customerId": undefined,
  "email": "test@test.com",
  "firstName": "test",
  "id": "V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTox",
  "language": "EN",
  "lastName": "test",
  "municipality": "helsinki",
  "phoneNumber": "555555",
  "queue": null,
  "status": "PENDING",
  "summaryInformation": Object {
    "acceptBoatingNewsletter": true,
    "acceptFitnessNews": true,
    "acceptLibraryNews": true,
    "acceptOtherCultureNews": true,
    "applicationCode": "1234",
  },
  "winterStorageAreaChoices": Array [
    Object {
      "__typename": "WinterStorageAreaChoiceType",
      "priority": 1,
      "winterStorageArea": null,
      "winterStorageAreaName": "Iso-Sarvasto",
    },
    Object {
      "__typename": "WinterStorageAreaChoiceType",
      "priority": 2,
      "winterStorageArea": null,
      "winterStorageAreaName": "Rajasaari",
    },
  ],
  "zipCode": "00100",
}
`;
